File: src/main/java/eterea/api/rest/model/VoucherProducto.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package eterea.api.rest.model;

import java.io.Serializable;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 *
 * @author Romina
 */
@Data
@Entity
@Table(name = "voucherproducto")
@EqualsAndHashCode(callSuper=false)
@NoArgsConstructor
@AllArgsConstructor
public class VoucherProducto extends Auditable implements Serializable{
    /**
	 * 
	 */
	private static final long serialVersionUID = -5539001630148258742L;

	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "VPr_ID")
    private Long voucherProductoId;

    @Column(name = "VPr_Vou_ID")
    private Long voucherId;
    
    @Column(name = "VPr_Prd_ID")
    private Integer productoId;
   
    @Column(name = "VPr_Paxs")
    private Integer cantidadPaxs;
         
}

File: src/main/java/eterea/core/api/rest/kotlin/model/VoucherProducto.kt
package eterea.core.api.rest.kotlin.model

import jakarta.persistence.*

@Entity
@Table(name = "voucherproducto")
data class VoucherProducto(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "VPr_ID")
    var voucherProductoId: Long? = null,

    @Column(name = "VPr_Vou_ID")
    var voucherId: Long? = null,

    @Column(name = "VPr_Prd_ID")
    var productoId: Int? = null,

    @Column(name = "VPr_Paxs")
    var cantidadPaxs: Int? = null,

    @OneToOne(optional = true)
    @JoinColumn(name = "vpr_prd_id", insertable = false, updatable = false)
    var producto: Producto? = null

) : Auditable() {
    data class Builder(
        var voucherProductoId: Long? = null,
        var voucherId: Long? = null,
        var productoId: Int? = null,
        var cantidadPaxs: Int? = null,
        var producto: Producto? = null
    ) {
        fun voucherProductoId(voucherProductoId: Long?) = apply { this.voucherProductoId = voucherProductoId }
        fun voucherId(voucherId: Long?) = apply { this.voucherId = voucherId }
        fun productoId(productoId: Int?) = apply { this.productoId = productoId }
        fun cantidadPaxs(cantidadPaxs: Int?) = apply { this.cantidadPaxs = cantidadPaxs }
        fun producto(producto: Producto?) = apply {this.producto = producto}
        fun build() = VoucherProducto(voucherProductoId, voucherId, productoId, cantidadPaxs, producto)
    }
}
